### DISCUSSION TITLE
Does shell integration support NuShell?

### DISCUSSION THREAD

**Discussion by gutenye** [kovidgoyal/kitty#7187] [Q&A] [answered]
It's not mentioned in the [doc](https://sw.kovidgoyal.net/kitty/shell-integration/#integration-with-other-shells), and a quick search in kitty and nutshell repos finds nothing.

If not supported, what needs to be done?

**Comment by kovidgoyal** [comment]
nushell implements some fraction of shell integration itself, I believe.
I dont know how complete it is. If you want to do it yourself you need
to figure out how to inject a shell script into nushell startup robustly
and in that shell script use hooks provided by the shell to mark the
prompt etc.

**Reply by gutenye** [reply]
Thanks. [Asked in the NuShell repo](https://github.com/nushell/nushell/discussions/12065)

**Reply by gutenye** [reply, marked as answer]
[NuShell supported it](https://github.com/nushell/nushell/discussions/12065#discussioncomment-8666333)
